ml-agents/mlagents/trainers/agent_processor.py

```diff
@@ -156,3 +156,49 @@ def add_experiences(
         self.policy.save_previous_action(
             curr_info.agents, take_action_outputs["action"]
         )
+
+    def publish_trajectory_queue(self, trajectory_queue: Queue) -> None:
+        """
+        Adds a trajectory queue to the list of queues to publish to when this AgentProcessor
+        assembles a Trajectory
+        :param trajectory_queue: Trajectory queue to publish to.
+        """
+        self.trajectory_queues.append(trajectory_queue)
+
+
+class AgentManagerQueue:
+    """
+    Queue used by the AgentManager. Note that we make our own class here because in most implementations
+    deque is sufficient and faster. However, if we want to switch to multiprocessing, we'll need to change
+    out this implementation.
+    """
+
+    def __init__(self, behavior_id: str = None):
+        """
+        Initializes an AgentManagerQueue. Note that we can give it a behavior_id so that it can be identified
+        separately from an AgentManager.
+        """
+        self.queue: Deque[Union[Trajectory, Policy]] = deque()
+        self.behavior_id = behavior_id
```

Comment:
Giving the queue a behavior_id is understandably a bit weird. I did it this way so that the Trainer can retrieve which behavior id the policy queue belongs to and publish the right policy to it. 

The alternative is to give the Trainer access to the AgentManager itself, but I wanted to keep the AgentManager abstraction separate from the Trainer. I'd imagine in the future, we'd want to assemble queues and Trainers in a different way.